lib/puppet/parser/functions/getvar_emptystring.rb

```diff
@@ -0,0 +1,35 @@
+module Puppet::Parser::Functions
+
+  newfunction(:getvar_emptystring, :type => :rvalue, :doc => <<-'ENDHEREDOC') do |args|
+    Lookup a variable in a remote namespace.
+    Return an empty string rather than undef/nil if it could not be found
+
+    For example:
+
+        $foo = getvar('site::data::foo')
+        # Equivalent to $foo = $site::data::foo
+
+    This is useful if the namespace itself is stored in a string:
+
+        $datalocation = 'site::data'
+        $bar = getvar("${datalocation}::bar")
+        # Equivalent to $bar = $site::data::bar
+    ENDHEREDOC
+
+    unless args.length == 1
+      raise Puppet::ParseError, ("getvar_emptystring(): wrong number of arguments (#{args.length}; must be 1)")
+    end
+
+    begin
+      result = self.lookupvar("#{args[0]}")
+
+      result = '' if result.nil?
```

Comment:
This is not nice - it translates puppet `undef` to empty string (the result if you render the `undef` as a string. IMO this brings back the same kind of old confusion over `undef` `:undef` and empty-string.